zaza/openstack/charm_tests/vault/utils.py

```diff
@@ -280,6 +281,65 @@ def ensure_secret_backend(client):
         pass
 
 
+def wait_until_ready(clients):
+    """Wait until vault is ready (for all clients).
+
+    Raise an error if it's not ready within a reasonable time.
```

Comment:
Okay, we're reaching end-game reviews now!  :)

Let's be specific with the docstrings please:

1. define what 'ready' means in this function.  e.g. has a `leader_address` and has an `ha_enabled` attribute on the clients' `ha_status`.
2. Poll for up to 2 minutes for this status to be achieved.
3. Raise an AssertionError exception if the status is not achieved for all the clients passed.

docstrings are for people to read to understand what the intent of the function is and what they can expect without reading the code.